core/src/main/scala/org/apache/spark/deploy/history/HistoryPage.scala

```diff
@@ -146,6 +202,69 @@ private[spark] class HistoryPage(parent: HistoryServer) extends WebUIPage("") {
       <td sorttable_customkey={info.lastUpdated.toString}>{lastUpdated}</td>
     </tr>
   }
+  
+  private def getAttemptURI(attemptInfo: ApplicationHistoryInfo, 
+                            returnEmptyIfAttemptInfoNull: Boolean = true ) = {
+    if (attemptInfo.appAttemptId.equals("")) { 
+      if(returnEmptyIfAttemptInfoNull) {
+        attemptInfo.appAttemptId
+      } else {
+        HistoryServer.UI_PATH_PREFIX + s"/${attemptInfo.id}"
+      }
+   } else {
+     HistoryServer.UI_PATH_PREFIX + s"/${attemptInfo.id}" + "_" +  s"${attemptInfo.appAttemptId}"
+     }
+  }
+  
+  private def firstAttemptRow(attemptInfo : ApplicationHistoryInfo)  = {
+    val uiAddress = 
+      if (attemptInfo.appAttemptId.equals("")) {
+       attemptInfo.appAttemptId
+      } else {
+       HistoryServer.UI_PATH_PREFIX + s"/${attemptInfo.id}" + "_" +  s"${attemptInfo.appAttemptId}"
+      }
+              
+    val startTime = UIUtils.formatDate(attemptInfo.startTime)
+    val endTime = UIUtils.formatDate(attemptInfo.endTime)
+    val duration = UIUtils.formatDuration(attemptInfo.endTime - attemptInfo.startTime)
+    val lastUpdated = UIUtils.formatDate(attemptInfo.lastUpdated)
+    val attemptId = attemptInfo.appAttemptId
+       <td><a href={uiAddress}>{attemptId}</a></td>
```

Comment:
Does scala actually consider this line as a continuation of the previous line? The inline XML syntax has always looked super weird to me...